server/src/main/java/org/elasticsearch/index/mapper/BooleanFieldMapper.java

```diff
@@ -237,23 +277,53 @@ protected void parseCreateField(ParseContext context, List<IndexableField> field
             return;
         }
 
-        Boolean value = context.parseExternalValue(Boolean.class);
-        if (value == null) {
-            XContentParser.Token token = context.parser().currentToken();
-            if (token == XContentParser.Token.VALUE_NULL) {
-                if (fieldType().nullValue() != null) {
-                    value = fieldType().nullValue();
+        Boolean value = null;
+        try {
+            value = context.parseExternalValue(Boolean.class);
+            if (value == null) {
+                final XContentParser parser = context.parser();
+                XContentParser.Token token = parser.currentToken();
+                if (token == XContentParser.Token.VALUE_NULL) {
+                    if (fieldType().nullValue() != null) {
+                        value = fieldType().nullValue();
+                    }
+                } else {
+                    if (indexCreatedVersion.onOrAfter(Version.V_6_0_0_alpha1)) {
```

Comment:
when i grabbed master the other day this was included, and now its gone. Im not sure if i should keep the lenient parsing support.

Will remove if required.